src/libraries/System.IO.FileSystem/tests/LargeFileTests.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO.Tests;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.IO.FileSystem.Tests
+{
+    [OuterLoop]
+    [ActiveIssue("https://github.com/dotnet/runtime/issues/45954", TestPlatforms.Browser)]
+    [Collection("NoParallelTests")] // don't create multiple large files at the same time
+    public class LargeFileTests : FileSystemTest
+    {
+        [Fact]
+        public async Task ReadAllBytesOverLimit()
+        {
+            using FileStream fs = new (GetTestFilePath(), FileMode.Create, FileAccess.Write, FileShare.Read, 4096, FileOptions.DeleteOnClose);
+
+            foreach (long lengthOverLimit in new long[] { Array.MaxLength + 1L, int.MaxValue + 1L })
```

Comment:
Will need to get rid of this case because it's an edge case that isn't fixed in 6.0; see: https://github.com/dotnet/runtime/pull/61519.
```suggestion
            foreach (long lengthOverLimit in new long[] { int.MaxValue + 1L })
```